allow public key and contract addresses to be supplied for nim codex operation

michaelsbradleyjr opened the pull request:
Anticipates support for these cli parameters being added to nim-dagger.

---

The PR is in draft because I need some input from @Menduist whether this is the correct way to do what he suggested in https://github.com/status-im/dagger-contracts/issues/9#issuecomment-1094245717.

Looking at [infra-codex/blob/tanguybranch/ansible/group_vars/all.yml#L64-L68](https://github.com/status-im/infra-codex/blob/tanguybranch/ansible/group_vars/all.yml#L64-L68), my understanding is that values supplied there can be used to override values for the same-named variables in infra-role-nim-codex. Is that correct?

If so, after doing the one-off deployment discussed in https://github.com/status-im/dagger-contracts/issues/9, we'd want the changes in this PR plus a PR for infra-codex where the contract addresses are supplied below L64-L68 linked above. Is that correct?

We can discuss it here or in https://github.com/status-im/dagger-contracts/issues/9, but we need to think about the token contract in relation to the storage contract, and the additional parameters for the storage contract. When testing against my local POA network I used `TestToken`'s address and other `Storage` constructor params from the tests for dagger-contracts, but maybe we want something different for our infra-codex testnet?

Then there's the matter of modifying nim-dagger to make use of the values supplied via cli params.

michaelsbradleyjr committed:
allow public key and contract addresses to be supplied for nim codex operation

anticipates support for these cli parameters being added to nim-dagger

Menduist commented:
Looks correct in ansible terms, just a note
codex_eth_public_key should be optional (so inside a if in the j2)
The other two should be mandatory (https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#forcing-variables-to-be-defined)

I'll let @markspanbroek comment on the token contract relation :)

markspanbroek commented:
TestToken is ok for now. It'll allow everyone to mint their own tokens, which is convenient in a test setup.

Using the other parameters from [deploy/storage.js](https://github.com/status-im/dagger-contracts/blob/6aa2894521faa53a9896e800caf713499b33c318/deploy/storage.js) is ok, except for proofPeriod and proofTimeout. We'll need something more than 10 seconds and 5 seconds. I'd go with 10 minutes (600 seconds), and 5 minutes (300 seconds) for now. I expect that we'll need to fiddle with the parameters a bit to find optimal values, later on.

jakubgs commented:
What if there's no value for those flags? Shouldn't they be skipped? Or is an empty string a valid value for those flags?

Should the not be conditionally added only if value is specified?

michaelsbradleyjr commented:
> Shouldn't they be skipped?

`codex_eth_public_key` should be skipped, yes. I'll make the change as @Menduist suggested. The others we may not use, i.e. we're currently in favor of having a json file that provides those values, since we already have logic for that. However, we'll need to supply the json file, probably via infra-codex, but I'm not 100% sure.

michaelsbradleyjr committed:
[wip] conditionally supply codex_eth_public_key as cli param, drop contract vars/params

michaelsbradleyjr commented:
I added a wip commit, I think it's correct: `codex_eth_public_key` is supplied in a cli param only if it's not the empty string, and I dropped the vars/params for the contract addresses.

We may, though, want a var and cli param for the deployment json file.

jakubgs commented on defaults/main.yml:
```diff
@@ -31,6 +31,9 @@ codex_max_peers: 160
 codex_listening_port: 9000
 
+# Ethereum public key for Codex node
+codex_eth_public_key: ''
```
I'd rather comment it out and use `is defined` check.
```suggestion
#nim_codex_eth_public_key: ~
```
Also I added `nim_` prefix to variables.

jakubgs commented on templates/beacon-node.service.j2:
```diff
@@ -20,6 +20,9 @@ ExecStart={{ codex_repo_path }}/build/codex \
     --max-peers={{ codex_max_peers }} \
     --api-port={{ codex_rest_port }} \
     --cache-size={{ codex_cache_size }} \
+{% if codex_eth_public_key != "" %}
```
```suggestion
{% if nim_codex_eth_public_key is defined %}
```

michaelsbradleyjr committed:
Apply suggestions from code review

Co-authored-by: Jakub <i+github@always.fail>

jakubgs reviewed (approved):
Looks better, but it needs to be rebased, since I renamed all variables.

Menduist commented:
Stole your code in #2
